### DISCUSSION TITLE
OPC UA over TLS

### DISCUSSION THREAD

**Discussion by qorinator** [FreeOpcUa/opcua-asyncio#578] [Q&A] [answered]
Hi all.

I would like to have OPC UA messages over TLS 1.2. Is it possible to do it via this library? 

I saw in another [discussion](https://github.com/FreeOpcUa/python-opcua/issues/1077) that @AndreasHeine has mentioned SSL/TLS. however, beside the function set_security/set_security_string in Client class. I was not able to find anything else. 

Any help/guidance/a point to the right direction would be highly appreciated.
Thank you.

**Comment by AndreasHeine** [comment, marked as answer]
what do you mean with "over tls"? if you mean the HTTPS/SOAP stuff thats not implementet we currently support only the binary protocol with different encryption profiles!

**Reply by qorinator** [reply]
thank you for your response!

is this what you mean? the security policies 
![image](https://user-images.githubusercontent.com/13778492/119835056-c753df80-bf00-11eb-9262-2dc68f3005a1.png)
and the option to have sign or sign and encrypt? 

maybe another topic. I would like to implement a CA/certificate manager/certificate store. is there a way to configure the client to trust the CA and check server's certificate, if this certificate is issued by the CA?
